src/components/anchor-navigation/anchor-navigation.test.tsx

```diff
@@ -0,0 +1,364 @@
+    // (see https://github.com/testing-library/jest-dom/issues/461 - the variables seem to be
+    // treated by js-dom as "invalid values" as explained in the first reply) - so using toHaveStyleRule instead
+    SELECTED_ITEM_STYLES.forEach(([property, value]) => {
+      expect(screen.getAllByRole("listitem")[i]).toHaveStyleRule(
+        property,
+        value,
+        { modifier: "& a" }
+      );
+    });
+    expect(screen.getByTestId(`section-${i + 1}`)).toBeFocused();
+  }
+);
+
+test.each([0, 1, 2, 3, 4])(
+  "when Enter is pressed on a navigation item, the item is selected and the section container is focused",
+  async (i) => {
+    renderComponent();
+
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+    const anchorNavigationLinks = screen.getAllByRole("link");
+    anchorNavigationLinks[i].focus();
+    await user.keyboard("{Enter}");
+
+    act(() => {
+      jest.advanceTimersByTime(10);
+    });
+
+    SELECTED_ITEM_STYLES.forEach(([property, value]) => {
+      expect(screen.getAllByRole("listitem")[i]).toHaveStyleRule(
+        property,
+        value,
+        { modifier: "& a" }
+      );
+    });
+    expect(screen.getByTestId(`section-${i + 1}`)).toBeFocused();
+  }
+);
+
+test("does not alter the tabindex of the container if it was already focusable", async () => {
+  renderComponent();
+
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+  const anchorNavigationLinks = screen.getAllByRole("link");
+  anchorNavigationLinks[0].focus();
+  await user.keyboard("{Enter}");
+  act(() => {
+    jest.advanceTimersByTime(10);
+  });
+
+  expect(screen.getByTestId("section-1")).toHaveAttribute("tabindex", "0");
+});
+
+test("does nothing if a key other than tab or enter is pressed", async () => {
+  renderComponent();
+
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+  screen.getAllByRole("link")[1].focus();
+  await user.keyboard("{ArrowRight}");
+  act(() => {
+    jest.advanceTimersByTime(10);
+  });
+
+  SELECTED_ITEM_STYLES.forEach(([property, value]) => {
+    expect(screen.getAllByRole("listitem")[0]).toHaveStyleRule(
+      property,
+      value,
+      { modifier: "& a" }
+    );
+  });
+});
+
+test.each([
+  [399, 0],
+  [400, 1],
+  [799, 1],
+  [800, 2],
+  [1199, 2],
+  [1200, 3],
+  [1599, 3],
+  [1600, 4],
+  [1999, 4],
+  [2000, 4],
+])(
+  "scroll triggers selection of proper navigation item based on the navigation and sections top position",
+  (scrollPosition, selectedAnchorIndex) => {
+    renderComponent();
+
+    const topEdgeOffsets = [400, 800, 1200, 1600, 2000];
+    const refs = [ref1, ref2, ref3, ref4, ref5];
+
+    refs.forEach((ref, index) => {
+      if (ref.current) {
+        jest
+          .spyOn(ref.current, "getBoundingClientRect")
+          .mockImplementation(
+            () => ({ top: topEdgeOffsets[index] - scrollPosition } as DOMRect)
+          );
+      }
+    });
```

Comment:
If we're finding it hard to locate these elements using RTL queries, I wouldn't be opposed to `data-role`'s being added in a previous commit to help find any elements which can't be located otherwise.

Reply:
fair points - I'll see what I can do